### PR TITLE
adds tests for api core key.go

### PR DESCRIPTION
**Pull request by djosephsen**

### CODE REVIEW

**Comment by djosephsen** [comment]
this will close #62

File: api/core/key_test.go [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+				Type:   UINT64,
+				Uint64: 42,
+			},
+			want: "42",
+		},
+		{
+			name: `float32`,
+			v: Value{
+				Type:    FLOAT32,
+				Float64: 42,
```
Comment (rghetia) [inline api/core/key_test.go]: nit: can you please change 42 to 42.1? same with FLOAT64.

File: api/core/key_test.go [hunk 2]
```diff
@@ -0,0 +1,415 @@
+package core
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"go.opentelemetry.io/api/registry"
+	"testing"
+	"unsafe"
```
Comment (rghetia) [inline api/core/key_test.go]: please move "testing" and "unsafe" in separate block and on top.

File: api/core/key_test.go (hunk 1 above)
Comment (djosephsen) [reply api/core/key_test.go]: I did this initially, but after the type conversions the value came back as not exactly 42.1 but something like 42.100006542 😞 

42.0 works if that's acceptable

Comment (djosephsen) [reply api/core/key_test.go]: I suppose I could loosen the test to only consider one sig-fig 🤷‍♂

Comment (rghetia) [reply api/core/key_test.go]: I tried with 42.1 and I got 42.1
```
=== RUN   TestEmit/float32
have: 42.1
=== RUN   TestEmit/float64
have: 42.1
```

There are many test case in Opencensus that uses XY.Z form for double. Is it specific to your environment/setup. You may want to make the change and submit it and then see if the test passes/fails on circle-ci.

**Comment by jmacd** [comment]
Thanks @djosephsen, looks good.

File: api/core/key_test.go (hunk 1 above)
Comment (djosephsen) [reply api/core/key_test.go]: looks like it fails circle too 😞 

``` --- FAIL: TestFloat32 (0.00s)
    --- FAIL: TestFloat32/value:_float32(42.1) (0.00s)
        key_test.go:180:   core.Value{
              	... // 2 identical fields
              	Int64:   0,
              	Uint64:  0x00,
            - 	Float64: 42.1,
            + 	Float64: 42.099998474121094,
              	String:  "",
              	Bytes:   nil,
              }```

Comment (djosephsen) [reply api/core/key_test.go]: So it's only a single float test that's failing it, so I'm setting everything to 42.1 except TestFloat32(), which I'm setting to 42.0

Comment (rghetia) [reply api/core/key_test.go]: Earlier I had only changed it for TestEmit. It works there but it doesn't work for TestFloat32. So lets keep your original commit.

**Pull request merged**